content/doc/developer/testing/index.adoc

```diff
@@ -287,7 +287,112 @@ TODO: Write this section.
 You can create a Git repository during a test using `@GitSampleRepoRule`.
 
 === Using Agents
-TODO: Creating fake agents.
+Agents can be used in unitary tests creating a `DumbSlave` object to be used later in a Pipeline or in a Freestyle job.
+After that, the agent can be connected to the Controller using directly the JenkinsRule methods or retrieving the `Computer` object from the agent.
+
+[source,java]
+----
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.recipes.LocalData;
+
+import hudson.model.Computer;
+import hudson.model.FreeStyleProject;
+import hudson.model.labels.LabelAtom;
+import hudson.slaves.DumbSlave;
+import hudson.slaves.JNLPLauncher;
+
+public class WithAgentTest {
+
+  @Rule
+  public JenkinsRule jenkinsRule = new JenkinsRule();
+
+  @Test
+  @LocalData // Suppose you have already created a Freestyle and a WorkflowJob
+  public void test_agent() throws Exception {
+    // Creating the agent with an specific label
+    LabelAtom testingLabel = new LabelAtom("testing");
+    DumbSlave agent = jenkinsRule.createSlave(testingLablel);
+    /* If you need to change the Launcher
+    agent.setLauncher(new JNLPLauncher(true));
+    agent.save();
+    */
+    // Connecting the agent
+    jenkinsRule.waitOnline(agent);
+    /*
+    Alternative to waitOnline would be to use the Computer object, but you must check the agent become
+    online. Better use waitOnline
+
+    Computer computer = agent.toComputer();
+    while (!computer.isOnline()) {
+      computer.connect(true);
+    }
+    */
+
+    FreeStyleProject freeStyleProject = jenkinsRule.jenkins.getItemByFullName("my-freestyle", FreeStyleProject.class);
+    freeStyleProject.setAssignedLabel(testingLabel);
+    // Testing the job
+
+    WorkflowJob workflowJob = jenkinsRule.jenkins.getItemByFullName("my-workflow", WorkflowJob.class);
+    // Testing the job
+  }
+}
+----
+
+For the pipeline, just make sure the definition uses the label set in the test.
+For the example above, the pipeline might be
+
+[source,groovy]
+----
+pipeline {
+  agent {
+    label 'testing'
+  }
+
+  stages {
+    stage('Hello') {
+      steps {
+        sh 'echo Hello world'
+      }
+    }
+  }
+}
+----
+
+The agent can be disconnected as follows using the `OfflineCause` that fits better to the test.
+[source,java]
+----
+computer.disconnect(new OfflineCause.IdleOfflineCause());
+----
+
+==== Using Agents based in dockers
+Sometimes you might want to test an agent connected by SSH.
+For that purpose you can use the [docker-fixtures](https://github.com/jenkinsci/docker-fixtures) capability.
+Then the agent must use a `SSHLauncher`.
+
+[source,java]
+----
+public class WithAgentTest {
+
+  @Class
+  public DockerClassRule<JavaContainer> javaContainer = new DockerClassRule<>(JavaContainer.class);
```

Comment:
not sure if we want to be encouraging use of docker-fixtures, they are somewhat deprecated in favour of test-containers?

Reply:
I'm fine with removing this section. It was the first time I'd used agents in tests and I had to make quick research. I saw there were some tests using it, so I've included. I don't recall any use of test-containers in junit tests, but if you confirm me we don't want docker-fixtures to be used I'll happily remove it from the PR

Reply:
https://github.com/jenkinsci/docker-fixtures

 on the README it says:

> Probably this library should be deprecated in favor of [testcontainers.org](https://www.testcontainers.org/).

No one has gone through the work to remove from the eco system thought.

cc @jglick if any thoughts

Reply:
Docs should certainly not advertise use of `docker-fixtures`.

Testcontainers can be used to launch agent containers in various ways. You can create an inbound agent, though you then need to set the connection URL to use the Docker host, which is not very portable. Outbound SSH is possible using the standard `jenkins/ssh-agent` container, though it is a bit of a hassle to set up the SSH keypair; the image does not currently support simple password authentication, which would be more convenient for tests.

Reply:
Removed then